src/lib/actionables.js

```diff
@@ -0,0 +1,17 @@
+const Constants = require('../constants');
+
+module.exports = (ctx, cb) => {
+  const prevActionables = {
+    combine: null, uncombine: null, trash: null
+  }
+  return () => {
+    const nextActionables = cb();
+    if (nextActionables.uncombine === prevActionables.uncombine &&
+        nextActionables.combine === prevActionables.combine &&
+        nextActionables.trash === prevActionables.trash) {
+          return; // Nothing changed
+    }
+    ctx.map.fire(Constants.events.ACTIONABLE, nextActionables);
+    Object.assign(prevActionables, nextActionables);
+  }
+}
```

Comment:
Hmmm... I bit worried about doing a noop check for some actionable changes and not for others. But I think this is ok for now. We can think a bit more about how to deduplicate these events a bit more later on.